### PR TITLE
New version using cbapi-python bindings!

### PR DESCRIPTION
**Pull request by jgarman**

### CODE REVIEW

File: incident_report.py [hunk 1]
```diff
@@ -25,548 +25,164 @@
...
-        parts = netconn.split('|')
-        new_conn = {}
-        timestamp = datetime.strptime(parts[0], cb_datetime_format)
-        new_conn['timestamp'] = timestamp
+def copyStaticFiles(thedir ):
+    # append ./ to the directory unless it is already relative to the cwd
+    dirs_to_copy = ["css", "fonts", "js", "images"]
+    for d in dirs_to_copy:
+        fromdir = "./"+d
+        todir = "./"+thedir+"/"+d
```
Comment (jgarman) [inline incident_report.py]: change this to os.path.join() as well

File: incident_report.py [hunk 2]
```diff
@@ -25,548 +25,164 @@
...
+                     "filepaths": filepaths if filepaths else [] }
+
+    j2_env = Environment(loader=FileSystemLoader("."),
+                              trim_blocks=True)
 
-    def walk_executors_up(self, parent_guid, executors, depth = 10):
-        try:
-            if depth == 0:
-                return
+    htmlfile = file(outdir+"/index.html", 'wb')
```
Comment (jgarman) [inline incident_report.py]: same here - os.path.join()

File: incident_report.py [hunk 3]
```diff
@@ -25,548 +25,164 @@
...
+ 
+class IncidentReportGenerator(object):
+    
+    #constructor, DI the cbresponse api bindings object
+    def __init__(self, cbapi):
+        self.cb =  cbapi
+   
+    #generate a report from a target GUID
+    def generate_report(self, report_guid, verbose=False):
+        outdir="reports/"+report_guid
```
Comment (jgarman) [inline incident_report.py]: same here ^

File: incident_report.py [hunk 4]
```diff
@@ -25,548 +25,164 @@
...
+        process = self.cb.select(Process, report_guid)
+        
+        if (verbose):
```
Comment (jgarman) [inline incident_report.py]: `if verbose:`

File: incident_report.py [hunk 5]
```diff
@@ -25,548 +25,164 @@
...
-        parser.print_help()
-        sys.exit(-1)
-
-    rep = IncidentReport(opts.url, opts.token)
-    print "[+] Generating report for process guid: %s" % opts.guid
-    rep.generate_report(opts.guid)
-    print "[+] Report generated in ./%s" % rep.outdir + "/"
+        
+        #copy the static resources (images,css,js,etc) to the report directory
+        copyStaticFiles(outdir)
```
Comment (jgarman) [inline incident_report.py]: standardize on snake_case for variable names, CamelCase for class names

**Review by jgarman — COMMENTED** [review]
booboo

File: incident_report.py (hunk 1 above)
Comment (zacharyestep) [reply incident_report.py]: Done

File: incident_report.py (hunk 2 above)
Comment (zacharyestep) [reply incident_report.py]: Done

File: incident_report.py (hunk 4 above)
Comment (zacharyestep) [reply incident_report.py]: Done

File: incident_report.py (hunk 5 above)
Comment (zacharyestep) [reply incident_report.py]: Dome

**Comment by jgarman** [comment]
looks good, merging in - will prepare a new release of cbapi with you today

**Pull request merged**